int16[] target_id_list
